apps/web/src/pages/Teams/IndexTeamPage/components/TeamTable/helpers.tsx

```diff
@@ -102,7 +106,10 @@ export function roleAssignmentsToRoleTeamArray(
   return collection;
 }
 
-export function departmentAccessor(team: Team, intl: IntlShape) {
+export function departmentAccessor(
+  team: TeamTableTeamFragmentType,
+  intl: IntlShape,
+) {
```

Comment:
Lets you leverage the typecheck of the fragments downstream of component, ensuring functions operating on inputs have all the needed fields

Reply:
Such as, if you are missing departments from the query

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/9da0743d-d30f-4360-a685-6cd96d6a06f6)